drivers/serial/uart_stm32.c

```diff
@@ -1387,6 +1390,65 @@ static int uart_stm32_init(const struct device *dev)
 #endif
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int uart_stm32_set_power_state(const struct device *dev,
```

Comment:
This does not _exactly_ set the power state, which is confusing to future readers. I suggest moving the `data->pm_state = new_state` from below to here, or rename this function to something like `uart_stm32_prep_power_transition()`.

Reply:
However it seems to be more or less generic that the `uart_xx_pm_control` is calling a `uart_xx_set_power_state` function.
The test in the uart_stm32_set_power_state function is just checking that the UART is requested to enter a _non-DEVICE_PM_ACTIVE_STATE_

Reply:
Only `uart_nrfx_uart.c` is setting the field in device struct outside of `uart_*_set_power_state()`. The other two drivers which have PM enabled, `uart_cc13xx_cc26xx.c` and `uart_nrfx_uarte.c`, are doing this inside the setter function. I still think it is better to actually set the state inside the setter function to be less confusing.

Reply:
I'm personally still pro at least renaming the function if you want it to keep it separate.